src/app/organization/working-days/working-days.component.ts

```diff
@@ -0,0 +1,107 @@
+/** Angular Imports */
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder } from '@angular/forms';
+import { Router, ActivatedRoute } from '@angular/router';
+import { OrganizationService } from '../organization.service';
+
+/**
+ * Working-days component.
+ */
+@Component({
+  selector: 'mifosx-working-days',
+  templateUrl: './working-days.component.html',
+  styleUrls: ['./working-days.component.scss']
+})
+export class WorkingDaysComponent implements OnInit {
+
+  /** Working days data. */
+  workingDaysData: any;
+  /** Week days to be used for checkboxes */
+  weekDays = [
+    {name: 'MO', value: 'Monday', checked: false},
+    {name: 'TU', value: 'Tuesday', checked: false},
+    {name: 'WE', value: 'Wednesday', checked: false},
+    {name: 'TH', value: 'Thursday', checked: false},
+    {name: 'FR', value: 'Friday', checked: false},
+    {name: 'SA', value: 'Saturday', checked: false},
+    {name: 'SU', value: 'Sunday', checked: false}
+  ];
+  /** Working days form. */
+  workingDaysForm: FormGroup;
+  /** Data source for repayment reschedule type select */
+  selectOptions: any;
+
+  /**
+   * Retrieves the working days data from `resolve`.
+   * @param {formBuilder} formBuilder form builder.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {organizationService} organizationService organization service.
+   * @param {Router} router Router for navigation
+   */
+  constructor(private formBuilder: FormBuilder,
+              private route: ActivatedRoute,
+              private organizationService: OrganizationService,
+              private router: Router) {
+    this.route.data.subscribe((data: { workingDays: any }) => {
+      this.workingDaysData = data.workingDays;
+    });
+  }
+
+  /**
+   * Fill up the forms with the data from the server
+   */
+  ngOnInit() {
+    this.createWorkingDaysForm();
+  }
+
+  /**
+   * Creates the working days form.
+   */
+  createWorkingDaysForm() {
+    this.setWeekWorkingDays();
+    this.selectOptions = this.workingDaysData.repaymentRescheduleOptions;
+    const workingDaysFormData = {
+      'selectedOption': [this.workingDaysData.repaymentRescheduleType.id],
+      'extendTheTermForDailyRepayments': [this.workingDaysData.extendTermForDailyRepayments]
+    };
+    for (let i = 0; i < this.weekDays.length; i++) {
+      workingDaysFormData[this.weekDays[i].name] = this.weekDays[i].checked;
+    }
+    this.workingDaysForm = this.formBuilder.group(workingDaysFormData);
```

Comment:
Use formArray for recurrence in the form group, the format can be modified for the same in submit function. And define repaymentsRescheduledType and extendTermForDailyRepayments in the form group.